lib/include/jxl/decode.h

```diff
@@ -236,6 +236,33 @@ typedef enum {
   JXL_DEC_JPEG_RECONSTRUCTION = 0x2000,
 } JxlDecoderStatus;
 
+/** Rewinds decoder to the beginning. The same input must be given again from
+ * the beginning of the file and the decoder will emit events from the beginning
+ * again. When rewinding (as opposed to JxlDecoderReset), the decoder keeps
+ * state about the image, which it can use to skip to a requested frame more
+ * efficiently with JxlDecoderSkipFrames. After rewind,
+ * JxlDecoderSubscribeEvents must be used again, and it is feasible to leave out
+ * events that were already handled before, such as JXL_DEC_BASIC_INFO and
+ * JXL_DEC_COLOR_ENCODING, since they will provide the same information as
+ * before.
+ * @param dec decoder object
+ */
+JXL_EXPORT void JxlDecoderRewind(const JxlDecoder* dec);
+
+/** Makes the decoder skip the next `amount` frames. If this is used after
+ * JxlDecoderRewind to skip frames that were already decoded before, the decoder
+ * is able to skip more efficiently, as it can skip decoding earlier frames that
+ * it knows later frames don't depend on. A frame here is defined as a frame
+ * that without skipping emits events such as JXL_DEC_FRAME and JXL_FULL_IMAGE,
```

Comment:
What's the "current" frame here? I get that if I call SkipFrames(5) after Rewind() and then process the input I'll get the 6th frame pixels; but if I call Rewind() then process some events and call SkipFrames(5), what's the event that counts for the purpose of incrementing the concept of the current frame? JXL_DEC_FULL_IMAGE or JXL_DEC_FRAME?

Reply:
Added comment: if in the middle of processing a frame, it would only skip starting from the next one (this makes technically the most sense, as it already partially processed the current one)